app/components/filters/filter.ts

```diff
@@ -3,17 +3,19 @@ import RouterService from '@ember/routing/router-service';
 import { service } from '@ember/service';
 import { get } from '@ember/object';
 
-interface ArgsInterface {
-  id: string;
-  queryParam: string;
-  queryParamsA: string;
-  queryParamsB: string;
+interface Signature {
+  Args: {
+    id: string;
+    queryParam: string;
+    queryParamsA: string;
+    queryParamsB: string;
 
-  searchField: string;
-  options: Array<object>;
+    searchField: string;
+    options: Array<object>;
+  };
 }
 
-export default class FilterComponent extends Component<ArgsInterface> {
+export default class FilterComponent<S = Signature> extends Component<S> {
```

Comment:
I have no idea what I'm doing, but this seems to use the local `Signature` as a backup so that the other filter components don't fail typechecking.

Reply:
I meant subclassing our own components to share functionality that way.